### PR TITLE
[RFR] Response page

### PR DESCRIPTION
**Pull request by eschaefer**
- [x] Response route
- [x] Response form
- [x] Response error page
- [x] Response success page

Designs https://app.zeplin.io/project.html#pid=57e909ac2f96c44a785746d9&dashboard

- [x] Added Winner/Response tags to petition, if it is a winner or has a response
screen shot 2016-10-04 at 10 38 14

### CODE REVIEW

File: src/components/PetitionBody/index.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import settings from 'settings';
 import Heading2 from 'components/Heading2';
 import Paragraph from 'components/Paragraph';
 
-const PetitionBody = ({ description, suggestedSolution }) => (
+const PetitionBody = ({ description, suggestedSolution, cityResponse }) => (
```
Comment (mattberridge) [inline src/components/PetitionBody/index.js]: not sure if we should break this down into nested containers for

```
<PetitionDescription>
<PetitionSuggestedSolution>
<PetitionCityResponse>"
```

with a shared 'dumb' component for rendering headline / text?

think we're okay for now but any more and we should look to simplify.

Comment (eschaefer) [inline src/components/PetitionBody/index.js]: Yep, agreed.

**Comment by eschaefer** [comment]
Hey @sigmarstern, I can't get the `<RespondToPetitionForm>` to fire off the `respondToPetition` action. What am I missing?

File: src/routes.js [hunk 2]
```diff
@@ -36,6 +37,7 @@ export default function (props = {}) {
         <Route path='petitions/:id' component={Petition} />
         <Route path='petitions/:id/edit' component={EditPetition} />
         <Route path='petitions/:id/published' component={PublishedPetition} />
+        <Route path='respond/:token' component={RespondToPetition} />
```
Comment (mattberridge) [inline src/routes.js]: `PetitionResponse` ?

Comment (eschaefer) [inline src/routes.js]: That was my first thought, but I noticed the convention `[Verb]:Petition` kinda emerged.

**Comment by sigmarstern** [comment]
@eschaefer I noticed that `handleSubmit(respondToPetition)` is never being called. Perhaps due to failing validations?

File: src/components/RespondToPetitionForm/index.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+const RespondToPetitionForm = ({ petition, fields, handleSubmit, submitting }) => {
+  return (
+    <form onSubmit={handleSubmit(respondToPetition)}>
+      <Fieldset>
+        {FIELDS.map(field => (
+          <FormField
+            key={field.name}
+            config={field}
+            helper={fields[field.name]}
```
Comment (sigmarstern) [inline src/components/RespondToPetitionForm/index.js]: @mattberridge, @Tom-Bonnike, @eschaefer

This won't work with nested form fields like `answer.text` or `answer.name`. I managed to solve this using `helper={eval('fields.' + field.name)}` instead but this feels a bit hacky. Is there a better way to do this?

**Comment by sigmarstern** [comment]
How to test this?

1. Use [POST /petitions/{contentId}/event/force_state](http://api-iris-dev.lovelysystems.com/swaggerui#!/petition_event/post_petitions_contentId_event_force_state) to set a petition's state to `processing.waitForLetterResponse`
2. Use [GET /admin/petitions/{contentId}](http://api-iris-dev.lovelysystems.com/swaggerui#!/petition_admin/get) to get the petition's `response_token`
3. Go to [/respond/{response_token}](https://iris-frontend-dev-pr-113.herokuapp.com/respond/1oVYF)

File: src/components/RespondToPetitionForm/index.js (hunk 3 above)
Comment (eschaefer) [inline src/components/RespondToPetitionForm/index.js]: Just saw you changed it to use lodash `get` 👍

File: test/selectors/petitionResponse.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { assert } from 'chai';
+import getPetionResponse from 'selectors/petitionResponse';
+
+describe('petionResponse selector', () => {
```
Comment (sigmarstern) [inline test/selectors/petitionResponse.js]: 👍 @eschaefer

Comment (eschaefer) [inline test/selectors/petitionResponse.js]: 👯

**Comment by eschaefer** [comment]
@mattberridge @Tom-Bonnike What do you think about merging this PR before it grows bigger? Could open additional PRs related to Response.

**Pull request merged**